corehq/apps/es/client.py

```diff
@@ -442,6 +442,20 @@ def from_python(cls, doc):
         """
         raise NotImplementedError(f"{cls.__name__} is abstract")
 
+    @classmethod
+    def from_multi(cls, doc):
```

Comment:
What do you think of renaming `from_multi` to `from_dict` and have `from_python` do something like this?
```py
def from_python(cls, doc):
    return cls.from_dict(doc.to_json())
```

`from_dict` should not do any validation. It should assume that the dict being passed in is a valid data for this adapter. It is the responsibility of the caller to do validation if it is necessary in that context.

Reply:
I love it @millerdev. I will update the code accordingly.

>  My original understanding was that the from_multi method would just be an implementation detail on some adapter subclasses (only the ones that need to accept multiple types), and would get called from inside those adapter's from_python method. Thoughts?

I also thought of the same initially but given that all the adapters need this functionality so I moved it up in the parent class like `to_python`. Would love to offline on this and hash out the implementation details.